automation/label-approved-veps.py

```diff
@@ -5,106 +5,289 @@
+    variables = {"orgName": "kubevirt", "projectNumber": project_number}
+    has_next_page = True
+    current_cursor = None
+
+    print(
+        f"Fetching items from project 'kubevirt/projects/{project_number}' "
+        "to check for 'Status: Tracked'"
+    )
+
+    while has_next_page:
+        variables["cursor"] = current_cursor
+        result = execute_graphql_query(_TRACKED_ISSUES_QUERY, variables)
+
+        nodes, page_info = _extract_page_data_from_gql_result(
+                                result, project_number)
+
+        if nodes is None or page_info is None:
+            break
+
+        _process_page_items(nodes, tracked_issues)
+
+        has_next_page = page_info.get("hasNextPage", False)
+        current_cursor = page_info.get("endCursor") if has_next_page else None
+
+    return tracked_issues
+
+
 def main():
-    # Get PR body
+    """Main execution function."""
+    try:
+        project_number = parse_project_url(TARGET_PROJECT_URL)
+    except ValueError as e:
+        print(f"Error parsing project URL '{TARGET_PROJECT_URL}':\n{e}")
+        return
+
     pr_body = get_pr_details()
     if not pr_body:
         print("No PR body found.")
         return
 
-    # Extract enhancements reference numbers
-    ref_numbers = extract_enhancements_references(pr_body)
-    if not ref_numbers:
+    ref_numbers_str = extract_enhancements_references(pr_body)
+    if not ref_numbers_str:
         print("No enhancements references found.")
         return
 
-    # Check the first reference
-    ref_number = ref_numbers[0]
-    # search for merged PRs mentioning the issue
-    related_prs = find_related_merged_prs(ref_number)
-    if related_prs:
-        print(f"Found merged PR(s) for enhancements#{ref_number}: {related_prs}")
+    ref_numbers = {int(num_str) for num_str in ref_numbers_str}
+    print(
+        f"Referenced KubeVirt Enhancement issue(s) in PR "
+        f"{KUBEVIRT_REPO}/pulls/{PR_NUMBER}: {ref_numbers}"
+    )
+
+    tracked_issues_in_project = get_tracked_enhancement_issues_from_project(
+        project_number
+    )
+
+    first_matching_vep = None
+    for vep_issue_num in ref_numbers:
+        if vep_issue_num in tracked_issues_in_project:
+            first_matching_vep = vep_issue_num
+            break
+
+    if first_matching_vep is not None:
+        print(
+            f"Match: KubeVirt Enhancement issue #{first_matching_vep} "
+            f"(referenced in this PR) is tracked in project "
+            f"{TARGET_PROJECT_URL}."
+        )
+        print(
+            f"This PR ({KUBEVIRT_REPO}/pulls/{PR_NUMBER}) is related to a VEP "
+            "tracked for the current release."
+        )
         add_label_to_pr()
     else:
-        print(f"No merged PR found for enhancements#{ref_number}.")
+        print(
+            f"This PR ({KUBEVIRT_REPO}/pulls/{PR_NUMBER}) is not related to "
+            f"any VEP issue currently tracked in project {TARGET_PROJECT_URL}."
+        )
+
```

Comment:
I am wondering if we should cover the non-tracked ones.
For instance: Tracked ones that become Deferred -> The label should be removed (and probably a hold or something else should be added?)
In general, I think that the removal handling is missing

Reply:
@fossedihelm Thank you! Added.
Now the label will be removed from PRs tied to VEP, which is no longer tracked.